src/vs/workbench/contrib/tasks/browser/runAutomaticTasks.ts

```diff
@@ -119,35 +130,30 @@ export class RunAutomaticTasks extends Disposable implements IWorkbenchContribut
 		return { tasks, taskNames, locations };
 	}
 
-	public static async runWithPermission(taskService: ITaskService, storageService: IStorageService, notificationService: INotificationService, workspaceTrustManagementService: IWorkspaceTrustManagementService,
+	private async _runWithPermission(taskService: ITaskService, storageService: IStorageService, notificationService: INotificationService, workspaceTrustManagementService: IWorkspaceTrustManagementService,
 		openerService: IOpenerService, configurationService: IConfigurationService, workspaceTaskResult: Map<string, IWorkspaceFolderTaskResult>) {
-		const isWorkspaceTrusted = workspaceTrustManagementService.isWorkspaceTrusted;
-		if (!isWorkspaceTrusted || configurationService.getValue(ALLOW_AUTOMATIC_TASKS) === 'off') {
-			return;
-		}
 
 		const hasShownPromptForAutomaticTasks = storageService.getBoolean(HAS_PROMPTED_FOR_AUTOMATIC_TASKS, StorageScope.WORKSPACE, false);
-		const { tasks, taskNames, locations } = RunAutomaticTasks._findAutoTasks(taskService, workspaceTaskResult);
+		const { tasks, taskNames, locations } = this._findAutoTasks(taskService, workspaceTaskResult);
 
 		if (taskNames.length === 0) {
 			return;
 		}
-
 		if (configurationService.getValue(ALLOW_AUTOMATIC_TASKS) === 'on') {
-			RunAutomaticTasks._runTasks(taskService, tasks);
+			this._runTasks(taskService, tasks);
 		} else if (configurationService.getValue(ALLOW_AUTOMATIC_TASKS) === 'auto' && !hasShownPromptForAutomaticTasks) {
 			// by default, only prompt once per folder
 			// otherwise, this can be configured via the setting
 			this._showPrompt(notificationService, storageService, openerService, configurationService, taskNames, locations).then(allow => {
```

Comment:
If I'm reading this correctly it looks like we now only show the prompt on folder open in folders where there are automatic tasks. This removes the "context-aware" approach that we used to have: show the prompt when the user has just run a task. The context-aware approach was done to avoid contributing to the notification fatigue that we have upon opening a folder.

Reply:
My concern is that if I configure automatic tasks to happen on `folderOpen`, when I open the folder, i'd expect them to run or for me to be asked if they can run and would feel they're broken if it doesn't happen.